src/controllers/bookController.ts

```diff
@@ -1,29 +1,151 @@
+                userName: 'markSQLSA',
+                password: 'Nacho[7]Dip',
+            },
+        },
+    };
+
     constructor() {
         this.router = Router();
         this.router.get('/:id', this.getBook.bind(this));
+        this.router.get('/', this.getBooks.bind(this));
+        this.router.post(
+            '/:bookTitle/:copies/:ISBN/:authorFirst/:authorLast',
+            this.createBook.bind(this),
+        );
+    }
+
+    async getBook(req: Request, res: Response) {
+        const connection = new Connection(this.config);
+        let requestStatement = new String(); //Plz ignore
+        const output = [];
 
-        this.router.post('/', this.createBook.bind(this));
+        // Setup event handler when the connection is established.
+        connection.on('connect', async function (err) {
+            if (err) {
+                console.log('Error: ', err);
+                throw err;
+            }
+            let id = Number(req.params.id);
+            if (isNaN(id)) {
+                return res.status(400).json({ status: 'Invalid Id' });
+            } else {
+                const sqlRequest = `SELECT * FROM books WHERE id = ${id}`;
+                const databaseRequest = new TediousRequest(
+                    sqlRequest,
+                    function (err, _rowCount, rows) {
+                        if (err) {
+                            console.log(err);
+                            throw err;
+                        } else {
+                            if (rows.length == 0) {
+                                return res
+                                    .status(400)
+                                    .json({ status: 'book not found' });
+                            } else {
+                                for (let j = 0; j < rows[0].length; j++) {
+                                    output.push(rows[0][j].value);
+                                }
+                                return res.status(200).json(output);
+                            }
+                        }
+                    },
+                );
+                requestStatement = connection.execSql(databaseRequest);
+            }
+        });
+        connection.connect();
     }
 
-    getBook(req: Request, res: Response) {
-        // TODO: implement functionality
-        return res.status(500).json({
-            error: 'server_error',
-            error_description: 'Endpoint not implemented yet.',
+    async createBook(req: Request, res: Response) {
+        const connection = new Connection(this.config);
+        let requestStatement = new String(); //Plz ignore
+        const output = [];
+
+        // Setup event handler when the connection is established.
+        connection.on('connect', async function (err) {
+            if (err) {
+                console.log('Error: ', err);
+                throw err;
+            }
+            const parsedBookTitle = req.params.bookTitle.split('_').join(' ');
```

Comment:
This is a bit weird. It's possible to send spaces in the path using URL encoding, which would avoid this. (although really this should be sent on the body of the request as mentioned above, which makes it even easier)